### PR TITLE
(IN) expression at WHERE for DELETE

### PR DESCRIPTION
**Pull request by fmarchenko**
Sphinx DELETE syntax not support function (IN(col, values)). This change replace sphinxsearch function IN to SQL expression.

### CODE REVIEW

File: testproject/testapp/tests.py [hunk 1]
```diff
@@ -221,6 +222,17 @@ def testDelete(self):
...
 
+    def testDeleteWithIn(self):
+        expected = self.create_multiple_models()
+        random.shuffle(expected)
+        delete_ids = expected[:random.randint(1, len(expected) - 1)]
+        count_rows = self.model.objects.all().count()
+        qs = self.model.objects.filter(id__in=delete_ids).delete()
+        after_count = self.model.objects.all().count()
+        self.assertEqual(count_rows - len(delete_ids), after_count)
+        list(self.model.objects.order_by())
```
Comment (tumb1er) [inline testproject/testapp/tests.py]: This line checks that queryset is fetched correctly from db. Not necessary in this test, or replace with 

``` python
assertListEquals(models_left_after_delete, ...)
```

File: sphinxsearch/backend/sphinx/compiler.py [hunk 2]
```diff
@@ -221,7 +221,15 @@ class SQLInsertCompiler(compiler.SQLInsertCompiler, SphinxQLCompiler):
...
 class SQLDeleteCompiler(compiler.SQLDeleteCompiler, SphinxQLCompiler):
-    pass
+    def as_sql(self):
+        sql, params = super(SQLDeleteCompiler, self).as_sql()
+
+        # empty SQL doesn't need patching
+        if (sql, params) == ('', ()):
+            return sql, params
+
+        sql = re.sub(r'\(IN\((\w+),\s([\w\s\%,]+)\)\)', '\\1 IN (\\2)', sql)
```
Comment (tumb1er) [inline sphinxsearch/backend/sphinx/compiler.py:231]: This regex may replace IN somewhere else, not only in WHERE clause. I don't want to add more substitutions than actually necessary. 
I'll look for another variants later today.
UPD: looks like DeleteCompiler is very simple: it only renders DELETE and WHERE clauses.

Comment (fmarchenko) [inline sphinxsearch/backend/sphinx/compiler.py:231]: But (IN) in the DELETE can be seen only in the WHERE or i wrong?

File: testproject/testapp/tests.py (hunk 1 above)
Comment (fmarchenko) [inline testproject/testapp/tests.py]: I do not quite understand what's the problem with the test :-(

File: sphinxsearch/backend/sphinx/compiler.py (hunk 2 above)
Comment (tumb1er) [inline sphinxsearch/backend/sphinx/compiler.py:231]: You are right, I'm still looking a way to remove substitution

File: testproject/testapp/tests.py (hunk 1 above)
Comment (tumb1er) [inline testproject/testapp/tests.py]: First, using random leads to test failures that are very difficult to reproduce.
Second, test should not contain lines that are excess for test subject.
And third, add assert that delete() deletes only what it should, and skips what it shouldn't.

File: sphinxsearch/backend/sphinx/compiler.py (hunk 2 above)
Comment (tumb1er) [inline sphinxsearch/backend/sphinx/compiler.py:231]: :+1: I don't see any options without refactoring "extra where" in SphinxQuerySet.filter

**Comment by tumb1er** [comment]
Merged. Do you needa new version at PyPI?

**Comment by fmarchenko** [comment]
It would be nice

**Comment by tumb1er** [comment]
0.5.0

**Pull request merged**